From c60e3889b03c1dfdf0753bfed2ee96cfdd4db648 Mon Sep 17 00:00:00 2001
From: Vikram Dattu <vikram.dattu@espressif.com>
Date: Tue, 23 Apr 2024 11:58:09 +0530
Subject: [PATCH 1/2] Update generate-iot-credential.sh script

 - Existing script is outdated and cannot work with latest aws-cli. Fixed the same
---
 scripts/generate-iot-credential.sh | 26 ++++++++++++--------------
 1 file changed, 12 insertions(+), 14 deletions(-)

diff --git a/scripts/generate-iot-credential.sh b/scripts/generate-iot-credential.sh
index c09469e0b..86c73b211 100755
--- a/scripts/generate-iot-credential.sh
+++ b/scripts/generate-iot-credential.sh
@@ -2,6 +2,7 @@
 # You need to setup your aws cli first, because this script is based on aws cli.
 # You can use this script to setup environment variables in the shell which samples run on.
 # https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/how-iot.html
+
 thingName="webrtc_iot_thing"
 thingTypeName="webrtc_iot_thing_type"
 iotPolicyName="webrtc_iot_policy"
@@ -20,9 +21,9 @@ mkdir -p $certPath
 
 # Step 1: Create an IoT Thing Type and an IoT Thing
 # The following example command creates a thing type $thingTypeName
-aws --profile default  iot create-thing-type --thing-type-name $thingTypeName > $tmpPath/iot-thing-type.json
+aws --region us-east-1  iot create-thing-type --thing-type-name $thingTypeName > $tmpPath/iot-thing-type.json
 # And this example command creates the $thingName thing of the $thingTypeName thing type:
-aws --profile default  iot create-thing --thing-name $thingName --thing-type-name $thingTypeName > $tmpPath/iot-thing.json
+aws --region us-east-1  iot create-thing --thing-name $thingName --thing-type-name $thingTypeName > $tmpPath/iot-thing.json
 
 # Step 2: Create an IAM Role to be Assumed by IoT
 # You can use the following trust policy JSON for the iam-policy-document.json:
@@ -39,7 +40,7 @@ echo '{
    ]
 }' > $tmpPath/iam-policy-document.json
 # Create an IAM role.
-aws --profile default  iam create-role --role-name $iotRoleName --assume-role-policy-document file://$tmpPath/iam-policy-document.json > $tmpPath/iam-role.json
+aws --region us-east-1  iam create-role --role-name $iotRoleName --assume-role-policy-document file://$tmpPath/iam-policy-document.json > $tmpPath/iam-role.json
 
 # You can use the following IAM policy JSON for the iam-permission-document.json:
 echo '{
@@ -60,10 +61,10 @@ echo '{
         }
     ]
 }' > $tmpPath/iam-permission-document.json
-# Next, you must attach a permissions policy to the IAM role you created above. 
-aws --profile default iam put-role-policy --role-name $iotRoleName --policy-name $kvsPolicyName --policy-document file://$tmpPath/iam-permission-document.json > $tmpPath/iot-policy.json
+# Next, you must attach a permissions policy to the IAM role you created above.
+aws --region us-east-1 iam put-role-policy --role-name $iotRoleName --policy-name $kvsPolicyName --policy-document file://$tmpPath/iam-permission-document.json > $tmpPath/iot-policy.json
 # Next, create a Role Alias for your IAM Role
-aws --profile default  iot create-role-alias --role-alias $iotRoleAlias --role-arn $(jq --raw-output '.Role.Arn' $tmpPath/iam-role.json) --credential-duration-seconds 3600 > $tmpPath/iot-role-alias.json
+aws --region us-east-1  iot create-role-alias --role-alias $iotRoleAlias --role-arn $(jq --raw-output '.Role.Arn' $tmpPath/iam-role.json) --credential-duration-seconds 3600 > $tmpPath/iot-role-alias.json
 
 # You can use the following command to create the iot-policy-document.json document JSON:
 cat > $tmpPath/iot-policy-document.json <<EOF
@@ -88,17 +89,17 @@ cat > $tmpPath/iot-policy-document.json <<EOF
 }
 EOF
 # Now you can create the policy that will enable IoT to assume role with the certificate (once it is attached) using the role alias.
-aws --profile default iot create-policy --policy-name $iotPolicyName --policy-document file://$tmpPath/iot-policy-document.json
+aws --region us-east-1 iot create-policy --policy-name $iotPolicyName --policy-document file://$tmpPath/iot-policy-document.json
 
 # Step 3: Create and Configure the X.509 Certificate
 # Create the certificate to which you must attach the policy for IoT that you created above.
-aws --profile default  iot create-keys-and-certificate --set-as-active --certificate-pem-outfile $certPath/$iotCert --public-key-outfile $certPath/$iotPublicKey --private-key-outfile $certPath/$iotPrivateKey > $tmpPath/certificate
+aws --region us-east-1  iot create-keys-and-certificate --set-as-active --certificate-pem-outfile $certPath/$iotCert --public-key-outfile $certPath/$iotPublicKey --private-key-outfile $certPath/$iotPrivateKey > $tmpPath/certificate
 # Attach the policy for IoT (KvsCameraIoTPolicy created above) to this certificate.
-aws --profile default  iot attach-policy --policy-name $iotPolicyName --target $(jq --raw-output '.certificateArn' $tmpPath/certificate)
+aws --region us-east-1  iot attach-policy --policy-name $iotPolicyName --target $(jq --raw-output '.certificateArn' $tmpPath/certificate)
 # Attach your IoT thing (kvs_example_camera_stream) to the certificate you just created:
-aws --profile default  iot attach-thing-principal --thing-name $thingName --principal $(jq --raw-output '.certificateArn' $tmpPath/certificate)
+aws --region us-east-1  iot attach-thing-principal --thing-name $thingName --principal $(jq --raw-output '.certificateArn' $tmpPath/certificate)
 # In order to authorize requests through the IoT credentials provider, you need the IoT credentials endpoint which is unique to your AWS account ID. You can use the following command to get the IoT credentials endpoint.
-aws --profile default  iot describe-endpoint --endpoint-type iot:CredentialProvider --output text > $tmpPath/iot-credential-provider.txt
+aws --region us-east-1  iot describe-endpoint --endpoint-type iot:CredentialProvider --output text > $tmpPath/iot-credential-provider.txt
 # In addition to the X.509 cerficiate created above, you must also have a CA certificate to establish trust with the back-end service through TLS. You can get the CA certificate using the following command:
 curl --silent 'https://www.amazontrust.com/repository/SFSRootCAG2.pem' --output $certPath/$cacert
 
@@ -109,6 +110,3 @@ export AWS_IOT_CORE_CERT=$(pwd)"/"$certPath"/"$iotCert
 export AWS_IOT_CORE_PRIVATE_KEY=$(pwd)"/"$certPath"/"$iotPrivateKey
 export AWS_IOT_CORE_ROLE_ALIAS=$iotRoleAlias
 export AWS_IOT_CORE_THING_NAME=$thingName
-        
-
-
-- 
2.39.3 (Apple Git-145)

